### ISSUE TITLE
An EKS cluster is not imported into Rancher sometimes

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#28208]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**

- Create an EKS cluster (specify all the input values)
- Create 2 node groups for the cluster.
- Import this EKS cluster in rancher.
- Cluster is stuck in `Waiting` state
- Phase of the cluster

```
kubectl get ekscc -n cattle-global-data c-4wclb -o yaml | grep phase
        f:phase: {}
  phase: active
```
- Rancher logs:
```
2020/08/03 23:11:31 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-4wclb: ClusterUnavailable 503: cluster not found, requeuing
2020/08/03 23:11:33 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-4wclb: ClusterUnavailable 503: cluster not found, requeuing
2020/08/03 23:11:35 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-4wclb: ClusterUnavailable 503: cluster not found, requeuing
2020/08/03 23:11:40 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-4wclb: ClusterUnavailable 503: cluster not found, requeuin
```

**Expected Result:**
cluster should be imported successfully and should be seen as active

**Other details that may be helpful:**
- Able to import an EKS cluster with 1 nodegroup

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): master-head - latest- commit id: `30e77d238`
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Imported EKS
- Kubernetes version (use `kubectl version`):

```
1.16
```

**Comment by aaronRancher** [comment]
This is also seen in an imported EKS cluster with 1 node group.
Cluster is stuck in `waiting` state.

Logs:
```
2020/08/04 00:27:11 [ERROR] Failed syncing peers [{10.42.2.8 [10.42.1.9] true false}]: failed to start user controllers for cluster c-9vw8s: ClusterUnavailable 503: cluster not found
2020/08/04 00:27:11 [INFO] error in remotedialer server [400]: websocket: close 1006 (abnormal closure): unexpected EOF
2020/08/04 00:27:11 [ERROR] Failed to serve peer connection 10.42.0.6: read tcp 10.42.2.8:35150->10.42.0.6:443: read: connection reset by peer
2020/08/04 00:27:13 [INFO] Adding peer wss://10.42.0.7/v3/connect, 10.42.0.7
2020/08/04 00:27:13 [INFO] Stopping cluster agent for c-md4wn
2020/08/04 00:27:13 [ERROR] failed to start cluster controllers c-md4wn: context canceled
2020/08/04 00:27:13 [ERROR] Failed syncing peers [{10.42.2.8 [10.42.0.7 10.42.1.9] true false}]: failed to start user controllers for cluster c-9vw8s: ClusterUnavailable 503: cluster not found
2020/08/04 00:27:13 [INFO] Stopping cluster agent for c-md4wn
2020/08/04 00:27:13 [ERROR] failed to start cluster controllers c-md4wn: context canceled
2020/08/04 00:27:14 [INFO] Handling backend connection request [10.42.1.9]
2020/08/04 00:27:14 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-9vw8s: ClusterUnavailable 503: cluster not found, requeuing
```

**Comment by rmweir** [comment]
Rancher was unable to communicate with the EKS cluster. The EKS cluster that I observed had private and public access enabled. This means that traffic from the internet can communicate with the cluster and that resources that share a VPC will communicate privately. There were no networking rules to enable this communication. We were able to resolve this by adding the cluster security group to each node of the Rancher management cluster. Adding a rule to allow inbound traffic from a security group associated with the management cluster's nodes to the EKS cluster's security group would likely also work.
